samcli/lib/package/artifact_exporter.py

```diff
@@ -602,6 +602,27 @@ def export_metadata(self, template_dict):
 
         return template_dict
 
+    def apply_global_values(self, template_dict):
+        """
+        Takes values from the "Global" parameters and applies them to resources where needed for packaging.
+
+        This transform method addresses issue 1706, where CodeUri is expected to be allowed as a global param for
+        packaging, even when there may not be a build step (such as the source being an S3 file). This is the only
+        known use case for using any global values in the package step, so any other such global value applications
+        should be scoped to this method if possible.
+        """
+        for _, resource in self.template_dict["Resources"].items():
+
+            resource_type = resource.get("Type", None)
+            resource_dict = resource.get("Properties", None)
+
+            if "CodeUri" not in resource and resource_type == AWS_SERVERLESS_FUNCTION:
```

Comment:
The bug doesn't state, but `DefinitionUri` of `Globals->Api` also needs same treatment

Reply:
Will check this with the Layers issue as well.